umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/distributions_related_url.clj

```diff
@@ -23,6 +23,34 @@
   (let [value (subs str start (some #(when % %) args))]
     (str/trim (subs value (inc (.indexOf value ":"))))))
 
+(def description-string-field-re-pattern
+  "Returns the pattern that matches all the related fields in description-string"
+  (re-pattern "URLContentType:|Description:|Type:|Subtype:|Checksum:"))
+
+(defn- convert-description-string-to-map
+  "Convert Description string to a map, removing fields that are empty or nil.
+  Description string: \"URLContentType: DistributionURL Description: A very nice URL Type: GET DATA Subtype: Subscribe\"
+  Description map: {\"URLContentType\" \"DistributionURL\"
+                    \"Description\" \"A very nice URL\"
+                    \"Type\" \"GET DATA\"
+                    \"Subtype\" \"Subscribe\"}"
+  [description-string description-regex]
+  (let [description-string (-> description-string
+                               (str/replace description-regex
+                                            #(str "HSTRING" %1 "TSTRING"))
```

Comment:
What are HSTRING and TSTRING? We should probably pull these magic strings out into their own defined vars with a docstring.